Fix date picker field incorrect classname

talldan opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Note: This PR merges into the branch from #42890, not trunk

## Proposed changes:
Fixes an issue with the Date Picker field, that caused its input to have no padding.

This seems to be the case of the block using the wrong classname. In trunk the input has the `jetpack-field__input` classname. In #42890 it receives the `jetpack-field__date` class name.

This seems to be because the block's `type` isn't present in the `TEXT_FIELDS` array in the input block. Without that, `getInputClass` gives it a classname based on the type instead of the generic `jetpack-field__input` class:
https://github.com/Automattic/jetpack/blob/1c637437478656eddfe80fad3d7ca6c3d83777f0/projects/packages/forms/src/blocks/input/edit.js#L19-L29

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
1. Create a new post
2. Add a Jetpack form
3. Choose the 'Appointment' variation
4. Check that the date picker field in the resulting form has padding consistent with the other input fields.

talldan committed:
Ensure jetpack-field__input classname applies to date picker input instead of jetpack-field__date

aaronrobertshaw reviewed (commented):
Thanks for the attention to detail @talldan 👍 

I think there's a little more out of whack here. If you add a new date field it gets the `jetpack-field__input` classname the same as trunk.

When I recently added the variations, it looks like I mistakenly added the `type: 'date'` attribute to the Input block for the date picker fields in the variation inner block definitions. This was an oversight of mine to introduce the inconsistency.

I think we have two options:

1. Add the type attribute to the inner block's template definition in the `field-date` edit component, leaving the current proposed change in place.
2. Remove the type attribute from the contact form variation definitions of date fields and roll back the current proposed change.

The Input block's type isn't used in the editor on the rendered `input` element as they all need to be text fields to cover custom placeholders that might not match the type's format if that makes sense.

talldan commented:
I see. I think I lean towards the first option. Having the `type` defined makes it more consistent with the other field blocks.

talldan committed:
Remove explicit date type in appointment form variation

talldan committed:
Specify the field type for the date field in the block template

talldan commented:
I've pushed those changes, so hopefully this is good to merge 😄

aaronrobertshaw reviewed (commented):
It looks like we've flipped the inconsistency now i.e. removed the date type from the variation but added it to the inner blocks template. Shouldn't they be the same or is there something I'm missing?

To expand on the earlier review:

> 1. Add the type attribute to the inner block's template definition in the field-date edit component, leaving the current proposed change in place.

- Add the date type to the inner blocks template
- Leave the date type in the variations 
- Leave the inclusion of `date` in the `TEXT_FIELDS` array 

OR

> 2. Remove the type attribute from the contact form variation definitions of date fields and roll back the current proposed change.

- Leave the date type out of the inner blocks template
- Remove it from the variations
- Remove date from `TEXT_FIELDS`

I'll make some tweaks and re-test with those, then can push them here if you'd like.

aaronrobertshaw committed:
Iron out inconsistency of date type for date picker inputs

aaronrobertshaw committed:
Tied up text fields array

aaronrobertshaw reviewed (approved):
I went with the minimal set of changes and that brought the variations and inner blocks template inline.

✅ New additions of a date picker field have correct class and padding
✅ Date pickers in form variations such as the Appointment form have correct class

talldan commented:
> It looks like we've flipped the inconsistency now i.e. removed the date type from the variation but added it to the inner blocks template. Shouldn't they be the same or is there something I'm missing?

The majority of other inputs in the variation templates don't have the `type` specified, so that's why I removed it.

aaronrobertshaw commented:
Sounds like there's a potential follow up to ensure that all the other blocks are consistent between inner blocks templates and variations too.

talldan commented:
I also wonder whether the label/input inner blocks in the variation templates need to be specified at all.

It seems to work ok without them, the field blocks inserts its own default template.

Maybe there's something else that's broken when they're not specified that I'm missing.
